assets/IMAGE_PAIRS_GENERATOR.md

```diff
@@ -0,0 +1,115 @@
+
+We support exhaustive, sequential, and descriptor-based image pair generation:
+- **Exhaustive** matching compares every image pair, ensuring all possible matches are found but at a high computational cost.
+- **Sequential** matching only compares neighboring images in a sequence (according to their image ID), making it faster but missing potentially valuable non-adjacent image pairs.
+- **Descriptor-based** matching leverages global images descriptors to pre-filter and identify images pairs that have a higher probability of overlap, balancing efficiency with accuracy. This is discussed further in the [following section](#global-image-descriptors).
+
+
+```python
+class ImagePairsGenerator:
+    def __init__(self, retriever: RetrieverBase, global_descriptor: Optional[GlobalDescriptorBase] = None):
+        self._global_descriptor: Optional[GlobalDescriptorBase] = global_descriptor
+        self._retriever: RetrieverBase = retriever
+
+    def __repr__(self) -> str:
+        return f"""
+            ImagePairGenerator:
+                {self._global_descriptor}
+                {self._retriever}
+        """
+
+    def generate_image_pairs(
+        self, client: Client, images: List[Future], image_fnames: List[str], plots_output_dir: Optional[Path] = None
+    ) -> List[Tuple[int, int]]:
+        def apply_global_descriptor(global_descriptor: GlobalDescriptorBase, image: Image) -> np.ndarray:
+            return global_descriptor.describe(image=image)
+
+        descriptors: Optional[List[np.ndarray]] = None
+        if self._global_descriptor is not None:
+            global_descriptor_future = client.scatter(self._global_descriptor, broadcast=False)
+
+            descriptor_futures = [
+                client.submit(apply_global_descriptor, global_descriptor_future, image) for image in images
+            ]
+
+            descriptors = client.gather(descriptor_futures)
+
+        return self._retriever.get_image_pairs(
+            global_descriptors=descriptors, image_fnames=image_fnames, plots_output_dir=plots_output_dir
+        )
+```
+
+## Global Image Descriptors
+
+Global desriptors work similar to local feature desriptors except that these methods generate a single descriptor for each image. Distances between these global image descriptors can then be used as a metric for the expected "matchability" of the image pairs during the correspondence generation phase, where a threshold can be used to reject potentially dissimilar image pairs before conducting correspondence generation. This reduces the likelihood of matching image pairs with little to no overlap that could cause erroneous correspondences to be inserted into the optimization process in the back-end while also significantly reducing the runtimes as compared to exhaustive matching. 
```

Comment:
nit: I dont see much of a similarity here, the descriptors are actually very different. To the point that I would say "unlike feature descriptors that learn a local descriptor for each image patch/pixel, global descriptors generate a single descriptor for each image". The difference seems more important that any similarity (which would only be in the model architecture).